cli/pkg/credentialmanager/handler_linux.go

```diff
@@ -37,22 +37,23 @@ func NewCredentialManager() (cm *CredentialManager) {
 	if err != nil {
 		log.Fatal(err)
 	}
+	initChecks()
 	return &CredentialManager{apiTokenFile: dir + ".keptn", credsFile: dir + ".keptn-creds"}
 }
 
 // SetCreds stores the credentials consisting of an endpoint and an api token using pass or into a file in case
 // pass is unavailable.
-func (cm *CredentialManager) SetCreds(endPoint url.URL, apiToken string) error {
+func (cm *CredentialManager) SetCreds(endPoint url.URL, apiToken string, namespace string) error {
 	if _, err := os.Stat(passwordStoreDirectory); os.IsNotExist(err) {
 		fmt.Println("Using a file-based storage for the key because the password-store seems to be not set up.")
-
-		return ioutil.WriteFile(cm.apiTokenFile, []byte(endPoint.String()+"\n"+apiToken), 0644)
+		apiTokenFile := cm.apiTokenFile + "__" + kubeConfigFile.CurrentContext + "__" + namespace
```

Comment:
Can you please move that line into a common function to reduce duplications?

Reply:
Sure

Reply:
@bacherfl It's only for linux others wont needs as they are storing in credential manager rather than as a file, let me know if you still want me to define this as a function

Reply:
I have added the function for that, also rebased with the master